src/pkg/caendr/caendr/services/cloud/lifesciences.py

```diff
@@ -127,5 +137,58 @@ def update_all_linked_status_records(kind, operation_name):
       logger.warn(f"Unrecognized kind: {kind}")
       continue
 
+    # Only send a notification if the report's status has not been updated yet
+    # TODO: Should be able to remove kind check if all report notifications merged into one system
+    should_send_notification = done \
+      and status_record['status'] not in [TaskStatus.COMPLETE, TaskStatus.ERROR] \
+      and kind in [NemascanMapping.kind, HeritabilityReport.kind]
+    logger.debug(f'Should send notification for report {status_record.id}: {should_send_notification}. (done = {done}, kind = {kind}, current status = {status_record["status"]})')
+
+    # Update the report status
     status_record.set_properties(status=status)
     status_record.save()
+
+    # Send the notification, if applicable
+    # In theory, doing this after the database update should prevent duplicate emails
+    # For now, only send email notifications to admin users
+    if should_send_notification:
+      record_owner = status_record.get_user()
+      if record_owner is not None:
+        if 'admin' in record_owner['roles']:
+          logger.debug(f'Sending email notification for report {status_record.id} to {record_owner["email"]} (ID {record_owner.name}).')
+          email_result = send_result_email(status_record, status)
+        else:
+          logger.debug(f'Skipping email notification for report {status_record.id} for user {record_owner["email"]} (ID {record_owner.name}): user is not an admin.')
+      else:
+        logger.warn(f'Could not send email notification for report {status_record.id}: no user found. ({dict(status_record)})')
+
+      if email_result.status_code == 200:
+        logger.debug(f'Email sent successfully ({email_result.status_code}): {email_result.text}')
+      else:
+        logger.error(f'Email failed to send ({email_result.status_code}): {email_result.text}')
+
+
+
+def send_result_email(record, status):
+
+  response = requests.get(
+    f'{MODULE_SITE_BASE_URL}/api/notifications/job-finish/{record.kind}/{record.id}/{status}',
+    headers={
+      'Content-Type':  'application/json',
+      'Authorization': 'Bearer {}'.format(API_SITE_ACCESS_TOKEN),
+    },
+  )
+
+  # Get the JSON body from the request
+  if response.status_code != 200:
+    return response
+  message = response.json()
+
+  # Send the email
+  return send_email({
```

Comment:
if `send_mail()` returns any relevant info, make sure to log it. Also try/except?

Reply:
Fixed in [c1759eb](https://github.com/AndersenLab/CAENDR/pull/273/commits/c1759eb17cea4472307e2cd3817bb19fcad6c731)

Calling function does error handling / logging.